src/decoder/mod.rs

```diff
@@ -197,6 +230,18 @@ impl Wrapping for u64 {
     }
 }
 
+impl Wrapping for f32 {
+    fn wrapping_add(&self, other: Self) -> Self {
+        self + other
+    }
+}
+
+impl Wrapping for f64 {
+    fn wrapping_add(&self, other: Self) -> Self {
+        self + other
+    }
+}
```

Comment:
That's quite unusual, the assumption of predictors being that it is reversible. However, `f32` addition and subtraction is in fact lossy. Is this actually used in images?

Reply:
I'll have to look into this -- there is a third predictor for floating point I wasn't aware of. However libtiff also seems to accept horizontal predictor for float types (at least the horizontal predictor code doesn't check the format type) by just casting the bytes to uints. Haven't dug through the spec yet, but it's enough to block this PR for now.

Reply:
Found a specification, I think: http://chriscox.org/TIFFTN3d1.pdf

Reply:
Is this the only missing feature ? 
From the doc you link the `DecodeDeltaBytes` is the `rev_hpredict_nsamp` and the `DecodeFPDelta` could be implemented just by adding parameters to `rev_hpredict` and add some code to the current match. I am missing something ? 
Do we have some test files for this case ? All of my tiff are float with horizontal predictor.

Reply:
It certainly is not that easy. I wrote the comment to refer to PREDICTOR_FLOATINGPOINT. The standard horizontal predictor should work _on byte data_ and not on floats and that is not implemented correctly here either. Your file should be a fine test case for these.

Reply:
Also, I'm not quite sure where it's standardized. I'm somewhat grasping here because it integer expands samples for 4bit values and it would make sense to do something similar for floating point, i.e. reinterpret them. But I really can't find an exact reference for this.

Reply:
Supporting this theory that floats are reinterpreted as bytes is [this python tiff library](https://github.com/cgohlke/tifffile/blob/987083705d1d988da2f27a3fb40784963cb2dab9/tifffile/tifffile.py#L12115-L12133).